src/test/java/com/wb/WeatherForecastApiApplicationTests.java

```diff
@@ -0,0 +1,36 @@
+package com.wb;
+
+import static org.springframework.test.web.servlet.ResultMatcher.matchAll;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class WeatherForecastApiApplicationTests {
```

Comment:
Klasa zawiera testy integracyjne więc nazwałbym ją `WeatherForecastApiApplicationIntegrationTests`
O ile dobrze jest mieć testy integracyjne żeby sprawdzić czy wszystko działa z docelowym systemem który wołasz w serwisie, to potrzebne są tez szybsze testy jednostkowe. Musisz stworzyc klase testowa o tej samej nazwie i podobnymi testami ale dodatkowo musisz zamockować swój `WeatherService` tak żeby zwracał to samo i żebyś tutaj w `expected` mógł przetestować parsowanie JSONa. Np tak jak tu: ostatni fragment kodu przed Summary-  https://spring.io/guides/gs/testing-web/